dna/src/entry_type.rs

```diff
@@ -0,0 +1,135 @@
+use std::{
```

Comment:
Yeah, really good to move this out of sys_entry. I did that in a local spike yesterday when starting to entangle our core types into a new crate for the HDK.

But why did you put EntryType into the DNA crate?
I don't think it belongs in here any more as it belongs into hash_table/sys_entry..

I would suggest moving it into a new crate. I created a `core_types` crate.

Reply:
I moved it in holochain_dna, because I want to use `EntryType::has_valid_app_name()` in that crate and did want to create a dependency to holochain_core. 
A new `core_types` crate might make sense but it feels really out of scope for this PR.
I also would like us to have clarity on the purpose and dependency of a new crate before making one.